internal/clientcache/internal/daemon/status_handler.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/clientcache/internal/cache"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// RefreshTokenStatus is the status of a resource token
+type RefreshTokenStatus struct {
+	Age      time.Duration
+	LastUsed time.Duration
+}
+
+type ErrorStatus struct {
+	Error        string
+	LastReturned time.Duration
+}
+
+// ResourceStatus contains the status of a specific resource type contained in
+// the cache for a specific user.
+type ResourceStatus struct {
+	Name         string
+	Count        int
+	LastError    *ErrorStatus        `json:",omitempty"`
+	RefreshToken *RefreshTokenStatus `json:",omitempty"`
+}
+
+// AuthTokenStatus contains the status of an auth token tracked in the cache for
+// a specific user.
+type AuthTokenStatus struct {
+	Id                    string
+	KeyringReferences     int `json:",omitempty"`
+	KeyringlessReferences int `json:",omitempty"`
+}
+
+// UserStatus contains the status of a specific user tracked by the cache
+type UserStatus struct {
+	// The Id of the user this status is for
+	Id string
+	// The boundary address for this user
+	Address string
+	// The auth tokens used by this user to authenticate with the boundary instance
+	AuthTokens []AuthTokenStatus
+	// The resources tracked by the cache for this user
+	Resources []ResourceStatus
+}
+
+// StatusResult is the struct returned to status requests.
+type StatusResult struct {
```

Comment:
Given that there will never be anything marshaled as JSON do you even need the tags here at all?

Reply:
Oops, sorry, `,omitempty` not `-,omitempty`. But please fix the names :-)